src/modules/Elsa.Elasticsearch/HostedServices/ConfigureIndexRolloverHostedService.cs

```diff
@@ -1,30 +1,52 @@
-using Elsa.Elasticsearch.Scheduling;
-using Elsa.Jobs.Schedules;
-using Elsa.Jobs.Services;
+using Elastic.Clients.Elasticsearch;
+using Elsa.Elasticsearch.Options;
+using Elsa.Elasticsearch.Services;
 using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
 
 namespace Elsa.Elasticsearch.HostedServices;
 
-public class ConfigureIndexRolloverHostedService : IHostedService
+/// <summary>
+/// A recurring job that applies the rollover strategy every interval.
+/// </summary>
+public class ConfigureIndexRolloverHostedService : BackgroundService
 {
-    private readonly IJobScheduler _jobScheduler;
+    private readonly ElasticsearchClient _client;
+    private readonly IEnumerable<IElasticConfiguration> _configurations;
+    private readonly ElasticsearchOptions _options;
 
-    public ConfigureIndexRolloverHostedService(IJobScheduler jobScheduler)
+    /// <summary>
+    /// Constructor.
+    /// </summary>
+    public ConfigureIndexRolloverHostedService(ElasticsearchClient client, IEnumerable<IElasticConfiguration> configurations, IOptions<ElasticsearchOptions> options)
     {
-        _jobScheduler = jobScheduler;
+        _client = client;
+        _configurations = configurations;
+        _options = options.Value;
     }
 
-    public async Task StartAsync(CancellationToken cancellationToken)
+    /// <inheritdoc />
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
     {
-        var job = new ConfigureIndexRolloverJob();
-        var schedule = new CronSchedule
+        while (!stoppingToken.IsCancellationRequested)
         {
-            // At the beginning of every month
-            CronExpression = "0 0 1 * *"
-        };
-        
-        await _jobScheduler.ScheduleAsync(job, GetType().Name, schedule, cancellationToken: cancellationToken);
+            await Task.Delay(_options.RolloverInterval, stoppingToken);
```

Comment:
I think if we use time interval instead of cron schedule (which makes sense) then we don't need a hosted service. We can basically pass MaxAge condition parameter with time interval data to rollover api of elastic client. This can be done directly in ConfigureClient method of IElasticConfiguration.

Reply:
Oh, nice! Then that simplifies things significantly 👍🏻

Reply:
Done